matchers/parse/parse.go

```diff
@@ -34,7 +37,13 @@ var (
 
 // Matchers parses one or more matchers in the input string. It returns an error
 // if the input is invalid.
-func Matchers(input string) (labels.Matchers, error) {
+func Matchers(input string) (matchers labels.Matchers, err error) {
+	defer func() {
```

Comment:
This was copied from [PromQL](https://github.com/prometheus/prometheus/blob/dd00c74faf03401139df99a32631c29dba0e9ef8/promql/parser/parse.go#L327-L340):

```go
func (p *parser) recover(errp *error) {
	e := recover()
	switch _, ok := e.(runtime.Error); {
	case ok:
		// Print the stack trace but do not inhibit the running application.
		buf := make([]byte, 64<<10)
		buf = buf[:runtime.Stack(buf, false)]

		fmt.Fprintf(os.Stderr, "parser panic: %v\n%s", e, buf)
		*errp = errUnexpected
	case e != nil:
		*errp = e.(error)
	}
}
```

Reply:
PromQL prints just the first 64KB of stack trace. We can do the same if needed. I found the [original commit](https://github.com/prometheus/prometheus/commit/5279d50d92c7259b1a20e6978f871827bbdfcbe1) and there is no justification for it being 64KB or evidence of buffering issues when writing to stderr. Perhaps it could be to assist log ingestion but I can't imagine the stack trace ever being more than a couple KB?